### ISSUE TITLE
`dotnet <sln> add proj` should have good defaults for ProjectTypeGuid

### ISSUE THREAD

**Issue by TheRealPiotrP** [dotnet/cli#5131]
@jgoshi

**Comment by jgoshi** [comment]
https://github.com/dotnet/sdk/issues/522

**Comment by jgoshi** [comment]
Algorithm:
1. Evaluate the project and get ProjectTypeGuid
2. Evaluate the project and get DefaultProjectTypeGuid (depends on the work for the sdk mentioned above)
3. if neither is available, fail with an error message.

Note that ProjectTypeGuid could have multiple guids. Choose the last one.
guid.Split(‘;’).Last()

**Comment by jgoshi** [comment]
All the infrastructure on our end is in place (including commented out tests). The current behavior remains unchanged (we just add the CPS project type guid when we cannot figure out the correct type). Once the SDK change goes in to provide us with the DefaultProjectTypeGuid it should be a simple change to throw when ProjectTypeGuid and DefaultProjectTypeGuid is not set.

**Comment by jgoshi** [comment]
In the master branch we have support for C# and VB. F# support is still in progress (see the comments near the end of this PR https://github.com/Microsoft/msbuild/pull/1607)

Once F# support is available, do the following: Search our codebase for  "https://github.com/dotnet/sdk/issues/522". It should appear in the following three files.
1. src\dotnet\ProjectInstanceExtensions.cs - Follow the instructions in the comment. Basically we want to throw the GracefulException if ProjectTypeGuid or DefaultProjectTypeGuid is not defined.

2. test\dotnet-migrate.Tests\GivenThatIWantToMigrateSolutions.cs - Enable the ProjectTypeGuid check. Note that the code comment is wrong, we don't burn in the C# guid but rather use the appropriate one (C#, VB, or F#).

3. test\dotnet-sln-add.Tests\GivenDotnetSlnAdd.cs - Enable the two tests (one is a Fact and the other is a Theory) which now will test the project type guid.

**Comment by cwe1ss** [comment]
Will this also cover "Service Fabric Applications" (*.sfproj)? It seems like they are currently also added incorrectly.

**Comment by peterhuene** [comment]
I believe this is now addressed for F#: https://github.com/Microsoft/visualfsharp/blob/44fa027b308681a1b78a089e44fa1ab35ff77b41/src/fsharp/FSharp.Build/Microsoft.FSharp.NetSdk.props#L38 due to setting the default project GUID.  I've confirmed that adding an F# project to the solution results in the correct type GUID.

@cwe1ss It should work for any project type that sets `DefaultProjectTypeGuid` in their SDK.

However, we still need to address @jgoshi's comment above; we should be throwing `GracefulException` saying the project cannot be added to the solution because it doesn't have a `ProjectTypeGuids` set (see #7742; related bug reading the wrong property) or `DefaultProjectTypeGuid` coming from an SDK rather than defaulting to C# (that's just wrong behavior, period).

I'll see about re-enabling any needed tests as well.

@livarcocc, given that this is closely related to #7742, want me to take this issue from you?
